tika-parsers/src/main/java/org/apache/tika/parser/crypto/TSDParser.java

```diff
@@ -0,0 +1,361 @@
+		//log.info("TSDParser --> getSupportedTypes SUPPORTED_TYPES: " + SUPPORTED_TYPES);
+		return SUPPORTED_TYPES;
+	}
+	
+	@Override
+	public void parse(InputStream stream, ContentHandler handler,
+            		  Metadata metadata, ParseContext context) throws IOException, SAXException, TikaException {
+		
+		//log.info("TSDParser --> parse START!");
+		//log.info("TSDParser --> InputStream in: " + stream);
+		//log.info("TSDParser --> ContentHandler in: " + handler);
+		//log.info("TSDParser --> Metadata in: " + metadata);
+		//log.info("TSDParser --> ParseContext in: " + context);
+		
+		//Try to parse TSD File
+		List<TSDMetas> tsdMetasList = this.buildMetas(stream);
+		
+		Integer count = 1;
+		
+		for(TSDMetas tsdm: tsdMetasList) {
+			metadata.set(TSD_LOOP_LABEL + count + " - " + Metadata.CONTENT_TYPE, TSD_MIME_TYPE);
+			metadata.set(TSD_LOOP_LABEL + count + " - " + Metadata.DESCRIPTION, TSD_DESCRIPTION_VALUE);
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_PARSED_LABEL, tsdm.getParseBuiltStr());
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_PARSED_DATE, tsdm.getParsedDateStr() + " " + this.TSD_DATE_FORMAT);
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_DATE, tsdm.getEmitDateStr() + " " + this.TSD_DATE_FORMAT);
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_POLICY_ID, tsdm.getPolicyId());
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_SERIAL_NUMBER, tsdm.getSerialNumberFormatted());
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_TSA, tsdm.getTSAstr());
+	        metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_ALGORITHM, tsdm.getAlgorithmName());
+	        count++;
+		}
+        
+        //log.info("TSDParser --> parse metadata: " + metadata);
+        
+        XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
+        xhtml.startDocument();
+        xhtml.endDocument();
+        
+        //log.info("TSDParser --> parse END!");
+	}
+	
+	private List<TSDMetas> buildMetas(InputStream stream) {
+		
+		List<TSDMetas> tsdMetasList = new ArrayList<TSDMetas>();
+		
+        try {
+             
+        	 //log.info("TSDParser --> buildMetas START!");
+             CMSTimeStampedDataParser cmsTimeStampedData = new CMSTimeStampedDataParser(stream);
+             
+			 TimeStampToken[] tokens = cmsTimeStampedData.getTimeStampTokens();
+			 
+			 for (int i=0; i < tokens.length; i++) {
+				 
+				 TSDMetas tsdMetas = new TSDMetas(true,
+						 						  tokens[i].getTimeStampInfo().getGenTime(),
+						 						  tokens[i].getTimeStampInfo().getPolicy().getId(),
+						 						  tokens[i].getTimeStampInfo().getSerialNumber(),
+						 						  tokens[i].getTimeStampInfo().getTsa(),
+						 						  tokens[i].getTimeStampInfo().getHashAlgorithm().getAlgorithm().getId());
+				 
+				 //log.info("TSDParser --> buildMetas genTime: " + tsdMetas.getEmitDateStr());
```

Comment:
You could use DEBUG or TRACE level for such messages, it sometimes helps. Also, with SLF4J preferred logger usage will look like
```java
LOG.debug("TSDParser --> buildMetas genTime: {}", tsdMetas.getEmitDateStr());
```
to avoid implicitly calling `tsdMetas.getEmitDateStr().toString()` when corresponding level isn't logged. Maybe in this case it's irrelevant, but produces clearer and better code.